src/context.ts

```diff
@@ -134,7 +134,11 @@ export class Context {
 
   async run(tool: Tool, params: Record<string, unknown> | undefined) {
     // Tab management is done outside of the action() call.
-    const toolResult = await tool.handle(this, tool.schema.inputSchema.parse(params || {}));
+    const toolResult = await tool.handle(this, tool.schema.inputSchema.parse(params || {})).catch(e => {
+      if (e instanceof Error && /locator._generateLocatorString: Timeout .* exceeded/.test(e.message))
```

Comment:
Shouldn't this be in `generateLocator()` helper instead?

Reply:
ah, yes. that makes total sense.